subsys/bluetooth/host/gatt.c

```diff
@@ -1464,40 +1606,47 @@ static int gatt_indicate(struct bt_conn *conn, u16_t handle,
 	return gatt_send(conn, buf, gatt_indicate_rsp, params, NULL);
 }
 
-#if defined(CONFIG_BT_GATT_DYNAMIC_DB)
-struct sc_data {
-	u16_t start;
-	u16_t end;
-};
-
-static void sc_save(struct bt_gatt_ccc_cfg *cfg,
-		    struct bt_gatt_indicate_params *params)
+#if defined(CONFIG_BT_GATT_SERVICE_CHANGED)
+static void sc_save(u8_t id, bt_addr_le_t *peer, u16_t start, u16_t end)
 {
-	struct sc_data data;
-	struct sc_data *stored;
+	struct gatt_sc_cfg *cfg;
+	bool modified;
 
-	memcpy(&data, params->data, params->len);
-
-	data.start = sys_le16_to_cpu(data.start);
-	data.end = sys_le16_to_cpu(data.end);
+	cfg = find_sc_cfg(id, peer);
+	if (!cfg) {
+		/* Find and initialize a free sc_cfg entry */
+		cfg = find_sc_cfg(BT_ID_DEFAULT, BT_ADDR_LE_ANY);
+		if (!cfg) {
+			BT_ERR("unable to save SC: no cfg left");
+			return;
+		}
 
-	/* Load data stored */
-	stored = (struct sc_data *)cfg->data;
+		cfg->id = id;
+		bt_addr_le_copy(&cfg->peer, peer);
+	}
 
 	/* Check if there is any change stored */
-	if (!stored->start && !stored->end) {
-		*stored = data;
+	if (!cfg->data.start && !cfg->data.end) {
```

Comment:
Same readability comment here. It's easier to read if using the same expression.

Reply:
Done in latest push!